mantidproject/mantid
# D2B round the tube centre indices

**Description of work.**

The central pixel indices for D2B vertical tubes are stored in the IPF. Since the tube centre cannot be determined with better precision than the pixel size itself, these numbers have to be rounded. 

This PR does that and also fixes an off-by-one error in maximum tube height calculation.

The unit test is fixed correspondingly.

**To test:**

Run this script and visualise the output (with spectrum viewer). 
There should NOT be black spots (empty bins) around the central line, as one gets with the current master (see below).

```
out = PowderDiffILLDetScanReduction(Run='538998.nxs', Output2D=True, Output1D=False, InitialMask=0, FinalMask=0)
```

![screen shot 2018-06-29 at 17 18 50](https://user-images.githubusercontent.com/16865759/42100765-5eb91934-7bc1-11e8-9b73-510df1583987.png)

Test file is below. You should set the facility to ILL first, and chop off the .zip extension.

[538998.nxs.zip](https://github.com/mantidproject/mantid/files/2149952/538998.nxs.zip)




No issue. 

This feature has been added for this release, so no need for reporting the bug fix.

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
